sdcm/nemesis.py

```diff
@@ -3560,117 +3561,101 @@ def disrupt_remove_node_then_add_node(self):
         if self._is_it_on_kubernetes():
             raise UnsupportedNemesis("On K8S nodes get removed differently. Skipping.")
 
-        node_to_remove = self.target_node
-        up_normal_nodes = self.cluster.get_nodes_up_and_normal(verification_node=node_to_remove)
+        verification_node = self.target_node
+        self.log.debug(f"Verification node is {verification_node}")
 
-        with Nemesis.run_nemesis(node_list=up_normal_nodes, nemesis_label="RemoveNodeAddNode") as verification_node:
-            self._remove_node_add_node(verification_node=verification_node, node_to_remove=node_to_remove)
+        nodes_by_rack = defaultdict(list)
+        for node in self.cluster.data_nodes:
+            if node == verification_node:
+                continue
+            self.log.debug(f"Following nodes and racks are target. Node: {node}, Rack: {node.rack}")
+            nodes_by_rack[node.rack].append(node)
 
-    def _remove_node_add_node(self, verification_node, node_to_remove, remove_node_host_id=None):
-        # node_to_remove must be different than node
-        # node_to_remove is single/last seed in cluster, before
-        # it will be terminated, choose new seed node
-        num_of_seed_nodes = len(self.cluster.seed_nodes)
-        if node_to_remove.is_seed and num_of_seed_nodes < 2:
-            new_seed_node = random.choice([n for n in self.cluster.nodes if n is not node_to_remove])
-            new_seed_node.set_seed_flag(True)
-            self.cluster.update_seed_provider()
+        eligible_racks = {rack: nodes for rack, nodes in nodes_by_rack.items() if len(nodes) >= 2}
+        random_nodes = [random.choice(nodes) for nodes in eligible_racks.values()]
 
```

Comment:
you should mark such node with nemesis_running, so they could not be occupied by another nemesis